src/ui/component-repository/components/list/HubRegistrySwitcher.js

```diff
@@ -0,0 +1,118 @@
+import React, { useCallback, useEffect } from 'react';
+import { Row, Col, DropdownKebab, MenuItem, Icon } from 'patternfly-react';
+import { useSelector, useDispatch } from 'react-redux';
+import { FormattedMessage } from 'react-intl';
+
+import { fetchRegistries, setActiveRegistry, setFetchedBundleGroups, setFetchedBundlesFromRegistry } from 'state/component-repository/hub/actions';
+import { getRegistries, getSelectedRegistry } from 'state/component-repository/hub/selectors';
+import { ECR_LOCAL_REGISTRY_NAME } from 'state/component-repository/hub/reducer';
+import { setVisibleModal, setInfo } from 'state/modal/actions';
+import { ADD_NEW_REGISTRY_MODAL_ID } from 'ui/component-repository/components/list/AddNewRegistryModal';
+import { DELETE_REGISTRY_MODAL_ID } from 'ui/component-repository/components/list/DeleteRegistryModal';
+
+const DEFAULT_ECR_REGISTRY = {
+  name: ECR_LOCAL_REGISTRY_NAME,
+  url: '',
+};
+
+const HubRegistrySwitcher = () => {
+  const activeRegistry = useSelector(getSelectedRegistry);
+  const registries = useSelector(getRegistries);
+  const dispatch = useDispatch();
+
+  const handleRegistryChange = useCallback((registry) => {
```

Comment:
Not sure why you're using `useCallback()` here and not with the other handlers below.

Reply:
done. Wrapped others in `useCallback` too ✅